### PR TITLE
Terminal informer after tenant cluster removed

### LINKED ISSUE #162: The informer of tenant cluster do not ternimal after the tenant cluster was deleted
**What steps did you take and what happened:**
[A clear and concise description on how to REPRODUCE the bug.]
After the `virtualCluster` removed, we do not terminal the `informer`s for the tenant cluster, so get tons of such error message in the `cyncer`'s log:
```
E0708 02:43:16.204417       1 reflector.go:138] pkg/mod/k8s.io/client-go@v0.21.1/tools/cache/reflector.go:167: Failed to watch *v1.Pod: failed to list *v1.Pod: Get "https://[apiserver-svc.default-ef338f-vc-sample-1]:6443/api/v1/pods?resourceVersion=264&timeout=30s": dial tcp: lookup apiserver-svc.default-ef338f-vc-sample-1 on 10.96.0.10:53: no such host
E0708 02:43:16.240855       1 reflector.go:138] pkg/mod/k8s.io/client-go@v0.21.1/tools/cache/reflector.go:167: Failed to watch *v1.ServiceAccount: failed to list *v1.ServiceAccount: Get "https://[apiserver-svc.default-ef338f-vc-sample-1]:6443/api/v1/serviceaccounts?resourceVersion=233&timeout=30s": dial tcp: lookup apiserver-svc.default-ef338f-vc-sample-1 on 10.96.0.10:53: no such host
E0708 02:43:18.098273       1 reflector.go:138] pkg/mod/k8s.io/client-go@v0.21.1/tools/cache/reflector.go:167: Failed to watch *v1.Namespace: failed to list *v1.Namespace: Get "https://[apiserver-svc.default-ef338f-vc-sample-1]:6443/api/v1/namespaces?resourceVersion=143&timeout=30s": dial tcp: lookup apiserver-svc.default-ef338f-vc-sample-1 on 10.96.0.10:53: no such host
E0708 02:43:19.240601       1 reflector.go:138] pkg/mod/k8s.io/client-go@v0.21.1/tools/cache/reflector.go:167: Failed to watch *v1.Event: failed to list *v1.Event: Get "https://[apiserver-svc.default-ef338f-vc-sample-1]:6443/api/v1/events?resourceVersion=268&timeout=30s": dial tcp: lookup apiserver-svc.default-ef338f-vc-sample-1 on 10.96.0.10:53: no such host
```

/kind bug

### PR DESCRIPTION
**Pull request by vincent-pli**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #162

### CODE REVIEW

File: virtualcluster/pkg/util/cluster/cluster.go [hunk 1]
```diff
@@ -79,7 +79,9 @@ type Cluster struct {
 	// a flag indicates that the cluster cache has been synced
 	synced bool
 
-	stopCh chan struct{}
+	// stopCh chan struct{}
```
Comment (zhuangqh) [inline virtualcluster/pkg/util/cluster/cluster.go:82]: just remove it?

File: virtualcluster/pkg/util/cluster/cluster.go [hunk 2]
```diff
@@ -324,5 +329,6 @@ func (c *Cluster) SetKey(k string) {
 
 // Stop send a msg to stopCh, stop the cache.
```
Comment (zhuangqh) [inline virtualcluster/pkg/util/cluster/cluster.go:330]: please update this comment
both in line 298

File: virtualcluster/pkg/util/cluster/cluster.go [hunk 3]
```diff
@@ -131,8 +133,8 @@ func NewCluster(key, namespace, name, uid string, getter mccontroller.Getter, co
 		RestConfig: clusterRestConfig,
 		options:    o,
 		synced:     false,
-		stopCh:     make(chan struct{}),
-		context:    context.Background(),
+		// stopCh:     make(chan struct{}),
```
Comment (Fei-Guo) [inline virtualcluster/pkg/util/cluster/cluster.go:136]: Can you remove the stopCh?

**Review by Fei-Guo — COMMENTED** [review]
Thanks for catching this. This should be due to the recent refactoring. One nit.

**Pull request merged**